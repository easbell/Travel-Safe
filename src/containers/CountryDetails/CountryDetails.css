.details {
  background: #fff;
  opacity: .8;
  width: 70%;
  margin: auto;
  padding: 15px;
  border-radius: 5px;
}