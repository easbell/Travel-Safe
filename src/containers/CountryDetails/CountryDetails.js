import React from 'react';

export const CountryDetails = (props) => {
  console.log(props)
  return (
    <div>
      <h2>{props.name}</h2>
      <h4>Safety Rating: {props.rating} / 5</h4>
      <h4>Recommended Vaccines:</h4>
      <p>Most Travelers: {props.vaccines['most travelers']}</p>
      <p>Some Travelers: {props.vaccines['some travelers']}</p>
    </div>
  )
}

export default CountryDetails;