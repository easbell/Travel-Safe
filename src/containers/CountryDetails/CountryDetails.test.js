import React from 'react';
import { CountryDetails } from './CountryDetails';
import { shallow } from 'enzyme';

describe('CountryDetails', () => {
  it('should match snapshot', () => {
    const country = {
      id: "AF",
      name: "Afghanistan",
      rating: 5,
      vaccines:
      {"most travelers": ["Hepatitis A", "Typhoid", "Polio"],
      "some travelers": ["Hepatitis B", "Cholera", "Malaria", "Rabies", "Yellow Fever"]}
    }

    let wrapper = shallow(
      <CountryDetails 
        {...country}
        key={country.name}
      />)
      expect(wrapper).toMatchSnapshot();
  });
});