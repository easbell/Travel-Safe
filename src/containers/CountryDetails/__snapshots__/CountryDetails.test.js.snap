// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountryDetails should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CountryDetails
    id="AF"
    name="Afghanistan"
    rating={5}
    vaccines={
      Object {
        "most travelers": Array [
          "Hepatitis A",
          "Typhoid",
          "Polio",
        ],
        "some travelers": Array [
          "Hepatitis B",
          "Cholera",
          "Malaria",
          "Rabies",
          "Yellow Fever",
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Afghanistan
        </h2>,
        <h4>
          Safety Rating: 
          5
           / 5
        </h4>,
        <h4>
          Recommended Vaccines:
        </h4>,
        Array [
          <p>
            most travelers
            : 
            Hepatitis A
            Typhoid
            Polio
          </p>,
          <p>
            some travelers
            : 
            Hepatitis B
            Cholera
            Malaria
            Rabies
            Yellow Fever
          </p>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Afghanistan",
        },
        "ref": null,
        "rendered": "Afghanistan",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Safety Rating: ",
            5,
            " / 5",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Safety Rating: ",
          5,
          " / 5",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Recommended Vaccines:",
        },
        "ref": null,
        "rendered": "Recommended Vaccines:",
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": "Hepatitis A,Typhoid,Polio",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "most travelers",
            ": ",
            Array [
              "Hepatitis A",
              "Typhoid",
              "Polio",
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "most travelers",
          ": ",
          "Hepatitis A",
          "Typhoid",
          "Polio",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": "Hepatitis B,Cholera,Malaria,Rabies,Yellow Fever",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "some travelers",
            ": ",
            Array [
              "Hepatitis B",
              "Cholera",
              "Malaria",
              "Rabies",
              "Yellow Fever",
            ],
          ],
        },
        "ref": null,
        "rendered": Array [
          "some travelers",
          ": ",
          "Hepatitis B",
          "Cholera",
          "Malaria",
          "Rabies",
          "Yellow Fever",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Afghanistan
          </h2>,
          <h4>
            Safety Rating: 
            5
             / 5
          </h4>,
          <h4>
            Recommended Vaccines:
          </h4>,
          Array [
            <p>
              most travelers
              : 
              Hepatitis A
              Typhoid
              Polio
            </p>,
            <p>
              some travelers
              : 
              Hepatitis B
              Cholera
              Malaria
              Rabies
              Yellow Fever
            </p>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Afghanistan",
          },
          "ref": null,
          "rendered": "Afghanistan",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Safety Rating: ",
              5,
              " / 5",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Safety Rating: ",
            5,
            " / 5",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Recommended Vaccines:",
          },
          "ref": null,
          "rendered": "Recommended Vaccines:",
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": "Hepatitis A,Typhoid,Polio",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "most travelers",
              ": ",
              Array [
                "Hepatitis A",
                "Typhoid",
                "Polio",
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "most travelers",
            ": ",
            "Hepatitis A",
            "Typhoid",
            "Polio",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "Hepatitis B,Cholera,Malaria,Rabies,Yellow Fever",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "some travelers",
              ": ",
              Array [
                "Hepatitis B",
                "Cholera",
                "Malaria",
                "Rabies",
                "Yellow Fever",
              ],
            ],
          },
          "ref": null,
          "rendered": Array [
            "some travelers",
            ": ",
            "Hepatitis B",
            "Cholera",
            "Malaria",
            "Rabies",
            "Yellow Fever",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
